### PR TITLE
bump kind to 0.17.0

### PR DESCRIPTION
**Pull request by k4leung4**
Signed-off-by: Kenny Leung <kleung@chainguard.dev>

# Changes

- 🧹 bump kind to 0.17.0

/kind cleanup

Fixes #

**Release Note**

```release-note
NONE
```

**Docs**

```docs

```

### CODE REVIEW

File: .github/workflows/kind-e2e.yaml [hunk 1]
```diff
@@ -20,9 +20,9 @@ jobs:
       fail-fast: false # Keep running if one leg fails.
       matrix:
         k8s-version:
-        - v1.23.12
-        - v1.24.6
-        - v1.25.2
+        - v1.23.13
```
Comment (nak3) [inline .github/workflows/kind-e2e.yaml:23]: Could you drop 1.23.x ?
It is no longer supported as https://github.com/knative/pkg/pull/2639 

The CI on this PR https://github.com/knative-sandbox/net-istio/pull/1009 could be fixed once we dropped 1.23.

Comment (k4leung4) [reply .github/workflows/kind-e2e.yaml]: done.

thanks for the pointing out the knative/pkg no longer support 1.23.x

**Pull request merged**